main.py

```diff
@@ -6,23 +6,32 @@
 # Global variable used for storing data (bad practice)
 data = []
 
+
 # Inefficient data manipulation
 @app.post("/add_item")
 def add_item(item: str):
     global data
     data.append(item)
     return {"message": "Item added successfully"}
 
+
 # Insecure API endpoint with missing authentication
 @app.get("/get_items")
 def get_items():
     return data
 
+
+@app.get("/get_item/{item_id}")
+def get_item(item_id: int):
+    return data[item_id]
```

Comment:
**Suggestion:** Add error handling for the case when the item_id is out of range in the get_item function.
```suggestion
@app.get("/get_item/{item_id}")
def get_item(item_id: int):
    if item_id < len(data):
        return data[item_id]
    else:
        raise HTTPException(status_code=404, detail="Item not found")

```